agent-connection/src/main/kotlin/com/luxoft/blockchainlab/corda/hyperledger/indy/AgentConnection.kt

```diff
@@ -77,29 +95,62 @@ class AgentConnection(val myAgentUrl: String, val invite: String? = null, val us
         val RESPONSE_RECEIVED = ADMIN_CONNECTIONS_BASE + "response_received"
 
         val STATE_REQUEST = ADMIN_BASE + "state_request"
+        val STATE_RESPONSE = ADMIN_BASE + "state"
     }
 
     private val webSocket = AgentWebSocketClient(URI(myAgentUrl))
 
+    fun checkState(stateMessage: State?, userName: String) : Boolean {
+        return if(stateMessage != null &&
+                stateMessage.content?.get("initialized") == true &&
+                stateMessage.content?.get("agent_name") == userName ) {
+            connectionStatus = ConnectionStatus.AGENT_CONNECTION_CONNECTED
+            true
+        } else {
+            connectionStatus = ConnectionStatus.AGENT_CONNECTION_DISCONNECTED
+            false
+        }
+    }
+
     init {
+        /**
+         * HTTP GET / in order to let the agent (python indy-agent) know its endpoint address
+         * indy-agent.py is incapable of determining its endpoint other than this way
+         */
+        val uri = URI(myAgentUrl)
+        val rootPath = "http://" + uri.host + ":" + uri.port + "/"
+        val rootUrl = URL(rootPath)
+        val res = rootUrl.openConnection().getInputStream()
+
         webSocket.apply {
             connectBlocking()
-            sendJson(WalletConnect(userName, passphrase))
-            if (invite != null) {
-                sendJson(ReceiveInviteMessage(invite))
-                val invite = waitForMessageOfType<InviteReceivedMessage>(INVITE_RECEIVED)
-                sendRequest(invite.key)
-                val response = waitForMessageOfType<RequestResponseReceivedMessage>(RESPONSE_RECEIVED)
-                counterParty = IndyParty(response.their_did, invite.endpoint)
+            sendJson(StateRequest())
+            var stateResponse = waitForMessageOfType<State>(STATE_RESPONSE)
+            if(!checkState(stateResponse, userName)) {
```

Comment:
i think we should keep only one approach for accepting invites.

Reply:
My vote would be to separate it from the constuctor then. Namely, to have a method acceptInvite() in the Connection interface. What would you think?

Reply:
Sounds good for me